cibuildwheel/platforms/android.py

```diff
@@ -0,0 +1,625 @@
+                    yield path, candidate
+            except ELFError:
+                # not an elf file
+                continue
+
+
+def soname_with_hash(src_path: Path) -> str:
+    """Return the same library filename as auditwheel would"""
+    shorthash = hashlib.sha256(src_path.read_bytes()).hexdigest()[:8]
+    src_name = src_path.name
+    base, ext = src_name.split(".", 1)
+    if not base.endswith(f"-{shorthash}"):
+        return f"{base}-{shorthash}.{ext}"
+    else:
+        return src_name
+
+
+# pylint: disable-next=too-many-positional-arguments
+def test_wheel(
+    config: PythonConfiguration,
+    build_options: BuildOptions,
+    build_path: Path,
+    python_dir: Path,
+    build_env: dict[str, str],
+    android_env: dict[str, str],
+    wheel: Path,
+) -> None:
+    if not (build_options.test_command and build_options.test_selector(config.identifier)):
+        return
+
+    log.step("Testing wheel...")
+    native_arch = arch_synonym(platform.machine(), platforms.native_platform(), "android")
+    if config.arch != native_arch:
+        log.warning(
+            f"Skipping tests for {config.arch}, as the build machine only supports {native_arch}"
+        )
+        return
+
+    if build_options.before_test:
+        shell_prepared(build_options.before_test, build_options=build_options, env=build_env)
+
+    # Install the wheel and test-requires.
+    site_packages_dir = build_path / "site-packages"
+    site_packages_dir.mkdir()
+    call(
+        "pip",
+        "install",
+        "--only-binary=:all:",
+        "--platform",
+        sysconfig_print("get_platform()", android_env).replace("-", "_"),
+        "--target",
+        site_packages_dir,
+        f"{wheel}{build_options.test_extras}",
+        *build_options.test_requires,
+        env=build_env,
+    )
+
+    # Copy test-sources.
+    cwd_dir = build_path / "cwd"
+    cwd_dir.mkdir()
+    if build_options.test_sources:
+        copy_test_sources(build_options.test_sources, Path.cwd(), cwd_dir)
+    else:
+        (cwd_dir / "test_fail.py").write_text(
+            resources.TEST_FAIL_CWD_FILE.read_text(),
+        )
+
+    # Android doesn't support placeholders in the test command.
+    if any(
+        ("{" + placeholder + "}") in build_options.test_command
+        for placeholder in ["project", "package"]
+    ):
+        msg = (
+            f"Test command '{build_options.test_command}' with a "
+            "'{project}' or '{package}' placeholder is not supported on Android, "
+            "because the source directory is not visible on the emulator."
+        )
+        raise errors.FatalError(msg)
+
+    # Parse test-command.
+    test_args = shlex.split(build_options.test_command)
+    if test_args[:2] in [["python", "-c"], ["python", "-m"]]:
+        test_args[:3] = [test_args[1], test_args[2], "--"]
+    elif test_args[0] in ["pytest"]:
+        test_args[:1] = ["-m", test_args[0], "--"]
```

Comment:
Let's add a warning here to match the iOS behaviour.

https://github.com/pypa/cibuildwheel/blob/5f22145df44122af0f5a201f93cf0207171beca7/cibuildwheel/platforms/ios.py#L630